docs/javascript/new-api_ajax.md

```diff
@@ -1,6 +1,111 @@
 # Ajax Requests - JavaScript API
 
-## Ajax inside Modules
+## `Promise`-based API for `DatabaseObjectAction`
+
+WoltLab Suite 5.5 introduces a new API for Ajax requests that uses `Promise`s
+to control the code flow. It does not rely on references to existing objects
+and does not use arbitrary callbacks to handle the setup and handling of the
+request.
+
+### Usage
+
+```ts
+// MyModule.ts
+import * as Ajax from "./Ajax";
+
+type ResponseGetLatestFoo = {
+  template: string;
+};
+
+export class MyModule {
+  private readonly bar: string;
+  private readonly objectId: number;
+
+  constructor(objectId: number, bar: string, buttonId: string) {
+    this.bar = bar;
+    this.objectId = objectId;
+
+    const button = document.getElementById(buttonId);
+    button?.addEventListener("click", (event) => this.click(event));
+  }
+
+  async click(event: MouseEvent): Promise<void> {
+    event.preventDefault();
+
+    const button = event.currentTarget as HTMLElement;
+    if (button.classList.contains("disabled")) {
+      return;
+    }
+    button.classList.add("disabled");
+
+    try {
+      const response = (await Ajax.dboAction("getLatestFoo", "wcf\\data\\foo\\FooAction")
+        .objectIds([this.objectId])
+        .payload({ bar: this.bar })
+        .dispatch()) as ResponseGetLatestFoo;
```

Comment:
```suggestion
      const response: ResponseGetLatestFoo = await Ajax.dboAction("getLatestFoo", "wcf\\\\data\\\\foo\\\\FooAction")
        .objectIds([this.objectId])
        .payload({ bar: this.bar })
        .dispatch();
```

I believe this works on an `unknown`, without needing to wrap this in parentheses.

When editing: Don't forget to escape the backslashes, when using the Jinja based codebox.

Reply:
`const response: ResponseGetLatestFoo` was rejected by the TS compiler (`ts2322`),  because it cannot assign `unknown` to this type.

Reply:
Ah, too bad. Disregard this then.